### PR TITLE
Convert tbe e2e to integration test

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
/kind cleanup

**What this PR does / why we need it**:
As of now, taint based eviction as an e2e test is causing troubles in different distros of kube because of inherent topology assumptions made like api-server availability, cloud provider, load balancer etc. This will be a problem in future considering that there is a chance that this test will get promoted to conformance assuming this feature GAs.

/cc @Huang-Wei @bsalamat @ahg-g 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/scheduling/taint_based_evictions.go [hunk 1]
```diff
@@ -1,195 +0,0 @@
...
-func getTolerationSeconds(tolerations []v1.Toleration) (int64, error) {
-	for _, t := range tolerations {
-		if t.Key == schedulerapi.TaintNodeUnreachable && t.Effect == v1.TaintEffectNoExecute && t.Operator == v1.TolerationOpExists {
-			return *t.TolerationSeconds, nil
-		}
-	}
-	return 0, errors.New("cannot find toleration")
-}
-
-var _ = SIGDescribe("TaintBasedEvictions [Serial]", func() {
```
Comment (bsalamat) [inline test/e2e/scheduling/taint_based_evictions.go]: Does this test any scheduler functionality? I think this should become a node test first and let SIG node decide whether an integration test make sense.

Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/taint_based_evictions.go]: Well, I think taintBasedEvictions have historically  been owned by sig-scheduling, even though the component which is responsible for this feature in nodelifecycle manager and the reason being we apply `NoExecute` taint and taints have been owned by sig-scheduling. However, you're right that it doesn't test scheduler functionality.

Comment (bsalamat) [reply test/e2e/scheduling/taint_based_evictions.go]: At some point it was, but I remember that @davidopp mentioned that it is now owned by SIG Node, which makes sense.

**Comment by BenTheElder** [comment]
> This will be a problem in future considering that there is a chance that this test will get promoted to conformance assuming this feature GAs.

It won't be promoted to conformance without being a reasonable test, that concern should not require any actions FWIW. Conformance tests have pretty tight requirements not to make assumptions like this.

File: test/integration/scheduler/taint_test.go [hunk 2]
```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
...
+		QPS:           -1,
+		Host:          context.httpServer.URL,
+		ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+	externalInformers := informers.NewSharedInformerFactory(externalClientset, time.Second)
+
+	admission.SetExternalKubeClientSet(externalClientset)
+	admission.SetExternalKubeInformerFactory(externalInformers)
+
+	// Apply feature gates to enable TaintNodesByCondition
+	algorithmprovider.ApplyFeatureGates()
```
Comment (Huang-Wei) [inline test/integration/scheduler/taint_test.go]:
```suggestion
	defer algorithmprovider.ApplyFeatureGates()()
```

Per #80144.

File: test/integration/scheduler/taint_test.go [hunk 3]
```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
...
+		v1.ResourceMemory: resource.MustParse("16Gi"),
+		v1.ResourcePods:   resource.MustParse("110"),
+	}
+	node1Taints := []v1.Taint{
+		{
+			Key:    schedulerapi.TaintNodeUnreachable,
+			Effect: v1.TaintEffectNoExecute,
+		}}
+
+	node0 := &v1.Node{
```
Comment (Huang-Wei) [inline test/integration/scheduler/taint_test.go]: Suggest to use a for loop for nodes initialization. Including creation.

Or, you can reuse `createNodes()`.

File: test/integration/scheduler/taint_test.go [hunk 4]
```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
...
+	// Update node condition.
+	err = updateNodeStatus(cs, node1)
+	if err != nil {
+		t.Fatalf("Cannot update node: %v", err)
+	}
+
+	if err := waitForNodeTaints(cs, node1, node1Taints); err != nil {
+		t.Errorf("Failed to taint node <%s>, err: %v", node1.Name, err)
+	}
+
```
Comment (Huang-Wei) [inline test/integration/scheduler/taint_test.go]: We should also implement the function included in original e2e tests:

- Default pods are applied tolerationSeconds=300, and evicted after 300 seconds
- Pod with tolerationSeconds=60 is evicted after 60 seconds
- Pods having tolerations matching unreachable are kept

And also remove the `Unknow` status to see if the taint is taken off from the node.

File: test/integration/scheduler/taint_test.go [hunk 5]
```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
...
+	cleanupNodes(cs, t)
+	waitForSchedulerCacheCleanup(context.scheduler, t)
+
```
Comment (Huang-Wei) [inline test/integration/scheduler/taint_test.go]: remove this line

File: test/integration/scheduler/taint_test.go [hunk 6]
```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
...
+}
+
+func TestTaintBasedEvictionForNodeNotReady(t *testing.T) {
```
Comment (Huang-Wei) [inline test/integration/scheduler/taint_test.go]: This function should be merged with TestTaintBasedEvictionsForNodeUnreachable using table testing.

FYI: https://github.com/kubernetes/kubernetes/blob/c68eeb230af92ea834eff70ba84c79e1c4c42dd9/test/integration/scheduler/predicates_test.go#L960-L966

**Review by Huang-Wei — CHANGES_REQUESTED** [review]
Thanks Ravi for bringing this up.

As we're replacing the e2e test with the integration one, we should ensure the function coverage remains the same. See my comments below.

**Pull request closed without merging**